### PR TITLE
#669 Automatically initialize Kerberos ticket if a keytab is provided

### PR DESCRIPTION
**Pull request by yruslan**

### CODE REVIEW

File: scripts/bash/run_enceladus.sh [hunk 1]
```diff
@@ -231,9 +231,26 @@ if [[ -z "$DRY_RUN" ]]; then
     DATE=`date +%Y_%m_%d-%H_%M_%S`
     NAME=`sed -e 's#.*\.\(\)#\1#' <<< $CLASS`
     TMP_PATH_NAME="$LOG_DIR/enceladus_${NAME}_${DATE}.log"
+
+    # Initializing Kerberos ticket
+    if [[ ! -z "$MENAS_AUTH_KEYTAB" ]]; then
+      # Get principle stored in the keyfile
+      PR=`printf "read_kt $MENAS_AUTH_KEYTAB\nlist" | ktutil | sed -n '5p' | awk '{print $3}' | cut -d '@' -f1`
```
Comment (lokm01) [inline scripts/bash/run_enceladus.sh]: 😲

File: scripts/bash/run_enceladus.sh [hunk 2]
```diff
@@ -170,6 +174,11 @@ validate "$DATASET_NAME" "--dataset-name"
 validate "$DATASET_VERSION" "--dataset-version"
 validate "$REPORT_DATE" "--report-date"
 
+if [[ "$DEPLOY_MODE" != "client" ]] && [[ "$DEPLOY_MODE" != "cluster" ]]; then
```
Comment (lokm01) [inline scripts/bash/run_enceladus.sh]: Is this meant to be checked against master instead?

**Comment by yruslan** [comment]
Added fixes to #674 to this PR as well since it affects the same file.

File: scripts/bash/run_enceladus.sh (hunk 2 above)
Comment (yruslan) [reply scripts/bash/run_enceladus.sh]: Indeed

Comment (yruslan) [reply scripts/bash/run_enceladus.sh]: Thanks a lot for spotting this, btw!

File: scripts/bash/run_enceladus.sh (hunk 1 above)
Comment (GeorgiChochov) [reply scripts/bash/run_enceladus.sh]: Jesus...

File: scripts/bash/run_enceladus.sh [hunk 3]
```diff
@@ -231,9 +240,26 @@ if [[ -z "$DRY_RUN" ]]; then
     DATE=`date +%Y_%m_%d-%H_%M_%S`
     NAME=`sed -e 's#.*\.\(\)#\1#' <<< $CLASS`
     TMP_PATH_NAME="$LOG_DIR/enceladus_${NAME}_${DATE}.log"
+
+    # Initializing Kerberos ticket
+    if [[ ! -z "$MENAS_AUTH_KEYTAB" ]]; then
+      # Get principle stored in the keyfile
+      PR=`printf "read_kt $MENAS_AUTH_KEYTAB\nlist" | ktutil | sed -n '5p' | awk '{print $3}' | cut -d '@' -f1`
```
Comment (Zejnilovic) [inline scripts/bash/run_enceladus.sh]: Is the `sed -n '5p'` failsafe way to go? on my mac, it seems it is on the 4th line. 
But I am not sure if a failsafe way exists. Could the bellow work on a larger scale? 
```bash
printf "read_kt $MENAS_AUTH_KEYTAB\nlist" | grep -Pio "(?<=\ )[A-Za-z0-9]*?(?=@) | head -1" 
```

Next thing is, do we want to throw an error if PR is empty?

File: scripts/bash/run_enceladus.sh [hunk 4]
```diff
@@ -170,6 +174,11 @@ validate "$DATASET_NAME" "--dataset-name"
 validate "$DATASET_VERSION" "--dataset-version"
 validate "$REPORT_DATE" "--report-date"
 
+if [[ "$MASTER" != "yarn" ]]; then
```
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply scripts/bash/run_enceladus.sh]: It has been explained to me that there is a default value.

**Pull request merged**